### PR TITLE
EOS-20349 : Unsupported features show/hide on UI

### PR DESCRIPTION
**Pull request by Vrishali-web-dev**
# UI
<pre>
  <code> 
Hide unsupported features on UI.
 </code>
</pre>
## Problem Statement
<pre>
  <code>
Story Ref (if any): EOS-20349 UI: unsupported features support on UI.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
No: Backend is not yet ready, hence tested this with mock data on the UI.
  </code>
</pre>
## Problem Description
<pre>
  <code>
- Based on list of features received from the API, show/hide components on UI.
  </code>
</pre>
## Solution/Screenshots
<pre>
  <code> 
- Added code to check the list of unsupported features and based on this components are show and hide on the UI.
- Created v-feature directive to check the support of feature on UI.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
CSM Login:
- Login to CSM UI
- Verify Capacity component in hidden
  </code>
</pre>
Capacity feature is hidden on the UI:

image

Signed-off-by: Vrishali Danave <vrishali.danave@seagate.com>

### CODE REVIEW

File: gui/src/services/api-register.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export default {
   sysconfig: `api/${version}/sysconfig`,
   bucket_policy: `/api/${version}/s3/bucket_policy`,
   user_permissions: `/api/${version}/permissions`,
+  unsupported_features: `/api/${version}/unsupportedfeatures`,
```
Comment (kupranay) [inline gui/src/services/api-register.ts:43]: is this API implemented in BE?

File: web/src/services/user/routes.ts [hunk 2]
```diff
@@ -91,5 +91,17 @@ export default [
         res.status(res.statusCode).send(result);
       }
     ]
+  },
+  {
+    path: "/api/:version/unsupportedfeatures",
```
Comment (kupranay) [inline web/src/services/user/routes.ts:96]: this API is not implemented in BE, this wil throw error

File: gui/src/components/dashboard/dashboard.vue [hunk 3]
```diff
@@ -24,7 +24,7 @@
       </v-col>
     </v-row>
     <v-row :style="{'min-height':alertTblRowHeightPx}">
-      <v-col
+      <v-col v-feature="capacity"
```
Comment (kupranay) [inline gui/src/components/dashboard/dashboard.vue:27]: capacity and all other features must be part of a constant file

Comment (kupranay) [reply gui/src/components/dashboard/dashboard.vue:27]: why only capacity. Where is the implementation of other Features?

**Review by kupranay — CHANGES_REQUESTED** [review]

**Comment by Vrishali-web-dev** [comment]
Implementation is different than expectation, new PR will be raised for the same and hence closing this PR.

**Pull request closed without merging**